src/main/java/io/confluent/connect/jdbc/sink/PoolOfPools.java

```diff
@@ -0,0 +1,77 @@
+package io.confluent.connect.jdbc.sink;
+
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.Maps;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.sql.DataSource;
+import java.util.Map;
+import java.util.Objects;
+
+public class PoolOfPools {
+
+  private static final Logger log = LoggerFactory.getLogger(PoolOfPools.class);
+  private static final Map<Key, DataSource> DATASOURCES = Maps.newConcurrentMap();
+
+  static class Key implements Comparable<Key> {
+    final String url;
+    final String username;
+    final String password;
+
+    Key(String url, String username, String password) {
+      this.url = url;
+      this.username = username;
+      this.password = password;
+    }
+
+    @Override
+    public int compareTo(Key that) {
+      return ComparisonChain.start()
+          .compare(this.url, that.url)
+          .compare(this.username, that.username)
+          .compare(this.password, that.password)
+          .result();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(this.url, this.username, this.password);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj instanceof Key) {
+        return 0 == compareTo((Key) obj);
+      } else {
+        return false;
+      }
+    }
+  }
+
+  static Key of(JdbcSinkConfig config) {
+    return new Key(config.getString(JdbcSinkConfig.CONNECTION_URL),
+        config.getString(JdbcSinkConfig.CONNECTION_USER),
```

Comment:
this could possibly fail if the authentication mechanism is not basic user-password validation.

Reply:
Look at the possible auth mechanisms supported by DB2 https://www.ibm.com/support/knowledgecenter/SSEPGG_11.1.0/com.ibm.db2.luw.admin.config.doc/doc/r0000294.html